Mathlib/Order/Interval/Finset/Nat.lean

```diff
@@ -193,6 +193,16 @@ theorem Ico_insert_succ_left (h : a < b) : insert a (Ico a.succ b) = Ico a b :=
   rw [Ico_succ_left, ← Ioo_insert_left h]
 #align nat.Ico_insert_succ_left Nat.Ico_insert_succ_left
 
+lemma Icc_insert_succ_left (h : a ≤ b) : insert a (Icc (a + 1) b) = Icc a b := by
+  ext x
+  simp
```

Comment:
Can you squeeze this `simp` by replacing it with the output of `simp?`? We want to avoid non-terminal `simp`'s because they are difficult to maintain.

Reply:
I didn't think of `omega` as a rigid tactic (thus, I didn't consider this to be one of the places where a non-terminal `simp` is actually a problem), but I've made that change.